lib/mappings.js

```diff
@@ -1,154 +1,190 @@
 'use strict';
 
-const Hoek = require('hoek');
+const Joi = require('joi');
 
 const internals = {};
-
-const maps = {
-    joiKnexMap: {
-        string        : 'string',
-        boolean       : 'boolean',
-        date          : 'date',
-        binary        : 'binary',
-        number        : 'float',
-        numberInteger : 'integer', // numberInteger is a value I created for Joi.number().integer()
-        array         : 'json',
-        object        : 'json',
-        any           : 'string'
-    },
-    knexJoiMap: {
-        string        : 'string',
-        boolean       : 'boolean',
-        date          : 'date',
-        binary        : 'binary',
-        float         : 'number',
-        integer       : 'numberInteger', // numberInteger is a value I created for Joi.number().integer()
-        json          : 'object' // or this could be an array... TODO
-    },
-    aliasMap: {
-
-        // Aliases across different database types
-
-        varchar   : ['character varying', 'varchar'],
-        boolean   : ['boolean', 'tinyint'],
-        integer   : ['integer', 'int'],
-        date      : ['date', 'datetime', 'timestamp with time zone'],
-        timestamp : ['timestamp'],
-        time      : ['time'],
-        json      : ['json'],
-        text      : ['text'],
-        float     : ['float', 'real'],
-        bigint    : ['bigint'],
-        decimal   : ['decimal'],
-        blob      : ['blob']
-    },
-    columnCompilerKnexMap: {
-        varchar   : 'string',
-        boolean   : 'boolean',
-        integer   : 'integer',
-        date      : 'date',
-        timestamp : 'timestamp',
-        time      : 'timestamp',
-        json      : 'json',
-        text      : 'string',
-        float     : 'float',
-        bigint    : 'integer',
-        tinyint   : 'integer',
-        decimal   : 'float',
-        blob      : 'binary'
-    }
+const maps = {};
+const convertFuncs = {};
+
+module.exports = { maps, convertFuncs };
+
+maps.joiKnexMap = {
+    string        : 'string',
+    boolean       : 'boolean',
+    date          : 'date',
+    binary        : 'binary',
+    number        : 'float',
+    numberInteger : 'integer', // numberInteger represents Joi.number().integer()
+    array         : 'json',
+    object        : 'json',
+    any           : 'string'
+};
+maps.knexJoiMap = {
+    string        : 'string',
+    boolean       : 'boolean',
+    date          : 'date',
+    timestamp     : 'date',
+    binary        : 'binary',
+    float         : 'number',
+    integer       : 'numberInteger', // numberInteger represents Joi.number().integer()
```

Comment:
If this felt odd, you could map to functions: `integer: (schema) => schema.number().integer()`